web/src/beta/features/PluginPlayground/Plugins/hook.ts

```diff
@@ -0,0 +1,188 @@
+    {
+      id: generateUniqueId(),
+      title: "My Plugin",
+      files: [REEARTH_YML_FILE]
+    }
+  ]);
+
+  const [selectedPluginId, setSelectedPluginId] = useState(plugins[0].id);
+  const [selectedFileId, setSelectedFileId] = useState<string>(
+    REEARTH_YML_FILE.id
+  );
+
+  const selectedPlugin = useMemo(
+    () =>
+      plugins.find((plugin) => plugin.id === selectedPluginId) ?? plugins[0],
+    [plugins, selectedPluginId]
+  );
+
+  const selectedFile = useMemo(
+    () =>
+      selectedPlugin.files.find((file) => file.id === selectedFileId) ??
+      selectedPlugin.files[0],
+    [selectedPlugin, selectedFileId]
+  );
+
+  const selectPlugin = useCallback((pluginId: string) => {
+    setSelectedPluginId(pluginId);
+  }, []);
+
+  const selectFile = useCallback((fileId: string) => {
+    setSelectedFileId(fileId);
+  }, []);
+
+  const addFile = useCallback(
+    (title: string) => {
+      const result = validateFileTitle(
+        title,
+        selectedPlugin.files.map((f) => f.title)
+      );
+      if (!result.success) {
+        return;
+      }
+      const newFile = {
+        id: generateUniqueId(),
+        title,
+        sourceCode: ""
+      };
+
+      setPlugins((plugins) =>
+        plugins.map((plugin) =>
+          plugin.id === selectedPlugin.id
+            ? { ...plugin, files: [...plugin.files, newFile] }
+            : plugin
+        )
+      );
+
+      setSelectedFileId(newFile.id);
+    },
+    [selectedPlugin]
+  );
+
+  const updateFileTitle = useCallback(
+    (newTitle: string, id: string) => {
+      const result = validateFileTitle(
+        newTitle,
+        selectedPlugin.files.map((f) => f.title)
+      );
+
+      if (!result.success) {
+        return;
+      }
+
+      setPlugins((plugins) =>
+        plugins.map((plugin) =>
+          plugin.id === selectedPlugin.id
+            ? {
+                ...plugin,
+                files: plugin.files.map((file) =>
+                  file.id === id ? { ...file, title: newTitle } : file
+                )
+              }
+            : plugin
+        )
+      );
+
+      return;
+    },
+    [selectedPlugin]
+  );
+
+  const deleteFile = useCallback(
+    (id: string) => {
+      setPlugins((plugins) =>
+        plugins.map((plugin) =>
+          plugin.id === selectedPlugin.id
+            ? {
+                ...plugin,
+                files: plugin.files.filter((file) => file.id !== id)
+              }
+            : plugin
+        )
+      );
+    },
+    [selectedPlugin]
+  );
+
+  const handleFileUpload = useFileInput((fileList) => {
+    const file = fileList?.[0];
+    if (!file) {
+      return;
+    }
+    const result = validateFileTitle(
+      file.name,
+      selectedPlugin.files.map((f) => f.title)
+    );
+    if (!result.success) {
+      return;
+    }
+
+    const reader = new FileReader();
+
+    reader.onload = async (e2) => {
+      const body = e2?.target?.result;
```

Comment:
is it okay if we use a more readable prop name like `event` or any other that might give more description to the prop?